modules/ROOT/pages/syntax/expressions.adoc

```diff
@@ -1261,6 +1261,20 @@ RETURN nullValue IS :: NULL AS isNull
 1+d|Rows: 1
 |===
 
+Note that closed dynamic types (`INNER_TYPE_1 | INNER_TYPE_2...`) cannot be appended generically with `NOT NULL`, instead all inner types must be nullable, or all not nullable. A syntax error will be raised if the inner types are not of the same nullability.
```

Comment:
Is closed dynamic types always refering to inner types?
How about adding a small why this is not allowed, e.g. "since null values could not be distinguished between the types"